lib/pushnotificationsservice.js

```diff
@@ -0,0 +1,380 @@
+        fs.stat(path.join(basePath, file), function(err, stats) {
+          return next(!err && stats.isDirectory());
+        });
+      }, function(dirs) {
+        return cb(null, dirs);
+      });
+    });
+  };
+
+  self.templatePath = path.normalize((opts.pushNotificationsOpts.templatePath || (__dirname + '/templates')) + '/');
+  self.defaultLanguage = opts.pushNotificationsOpts.defaultLanguage || 'en';
+  self.defaultUnit = opts.pushNotificationsOpts.defaultUnit || 'btc';
+  self.subjectPrefix = opts.pushNotificationsOpts.subjectPrefix || '';
+  self.publicTxUrlTemplate = opts.pushNotificationsOpts.publicTxUrlTemplate || {};
+  self.pushServerUrl = opts.pushNotificationsOpts.pushServerUrl;
+  async.parallel([
+
+    function(done) {
+      _readDirectories(self.templatePath, function(err, res) {
+        self.availableLanguages = res;
+        done(err);
+      });
+    },
+    function(done) {
+      if (opts.storage) {
+        self.storage = opts.storage;
+        done();
+      } else {
+        self.storage = new Storage();
+        self.storage.connect(opts.storageOpts, done);
+      }
+    },
+    function(done) {
+      self.messageBroker = opts.messageBroker || new MessageBroker(opts.messageBrokerOpts);
+      self.messageBroker.onMessage(_.bind(self._sendPushNotifications, self));
+      done();
+    },
+  ], function(err) {
+    if (err) {
+      log.error(err);
+    }
+    return cb(err);
+  });
+
+};
+
+PushNotificationsService.prototype._sendPushNotifications = function(notification, cb) {
+  var self = this;
+  cb = cb || function() {};
+
+  var notifType = PUSHNOTIFICATIONS_TYPES[notification.type];
+  if (!notifType) return cb();
+
+  // console.log(notification);
+
+  self._checkShouldSendNotif(notification, function(err, should) {
+    if (err) return cb(err);
+    if (!should) return cb();
+
+    self._getRecipientsList(notification, function(err, recipientsList) {
+      if (err) return cb(err);
+
+      async.waterfall([
+
+        function(next) {
+          self._readAndApplyTemplates(notification, notifType, recipientsList, next);
+        },
+        function(contents, next) {
+          async.map(recipientsList, function(recipient, next) {
+            var opts = {};
+            var content = contents[recipient.language];
+            opts.users = [notification.walletId + '$' + recipient.copayerId];
+            opts.android = {
+              "data": {
+                "title": content.plain.subject,
+                "message": content.plain.body
+              }
+            };
+            opts.ios = {
+              "alert": content.plain.body,
+              "sound": ""
+            };
+            return next(err, opts);
+          }, next);
+        },
+        function(optsList, next) {
+          async.each(optsList,
+            function(opts, next) {
+              self._makeRequest(opts, next());
+            },
+            function(err) {
+              if (err)
+                log.error(err);
```

Comment:
I don't think you need to log this error here. Logging at the final function is enough.